test/Middleware/ErrorHandlerTest.php

```diff
@@ -36,18 +37,17 @@ class ErrorHandlerTest extends TestCase
     /** @var MockObject&ResponseInterface */
     private $response;
 
-    /** @var callable():ResponseInterface */
-    private $responseFactory;
+    /** @var MockObject&ResponseFactoryInterface */
+    private ResponseFactoryInterface $responseFactory;
 
     private int $errorReporting;
 
     protected function setUp(): void
     {
+        $this->request         = $this->createMock(ServerRequestInterface::class);
         $this->response        = $this->createMock(ResponseInterface::class);
-        $response              = $this->response;
-        $this->responseFactory = static fn(): ResponseInterface => $response;
-
-        $this->request        = $this->createMock(ServerRequestInterface::class);
+        $this->responseFactory = $this->createMock(ResponseFactoryInterface::class);
+        $this->responseFactory->expects(self::any())->method('createResponse')->willReturn($this->response);
```

Comment:
We can remove the expectation and simplify a little bit, as it is set to "any":


```suggestion
        $this->responseFactory->method('createResponse')->willReturn($this->response);
```